compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -625,13 +626,23 @@ private JMethod resolveExactMethod(Class<?> clazz, String name, JMethodKind kind
 
         Executable executable = (kind == JMethodKind.CONSTRUCTOR) ? resolveConstructor(clazz, paramTypes) :
                 resolveMethod(clazz, name, paramTypes);
+        if (executable == null) {
```

Comment:
Another way we can do this is by changing validateArgumentTypes logic where we can skip first parameter if it is BalEnv class. But that has effect on normal interop function validation. This also looks good to me.

Reply:
I also consider that approach, I waned to support the case where people have put the `BalEnv` to the `paramTypes` manually as well, this works for both.